components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java

```diff
@@ -538,7 +539,7 @@ private void retriveSCIMUsers(List<Object> users, String[] userNames, List<Strin
                             + "Including user : " + userName + " in the response.");
                 }
                 User scimUser = this.getSCIMUser(userName, requiredClaims, scimToLocalClaimsMap);
-                if (scimUser != null) {
+                if (scimUser != null && StringUtils.isNotBlank(scimUser.getId())) {
```

Comment:
Shouldn't this be handled within the getSCIMUser it self similar to [1].
That's seems to be the core method being used to build up the user from SCIM schema and being referred in other places. Fixing only this place will not fix any other places where the same issue could occur

[1] https://github.com/wso2-extensions/identity-inbound-provisioning-scim2/blob/bb3cbb2d11fa4e2940c3fec29f662a0c21f1c21f/components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMUserManager.java#L2157

Reply:
Fixed